libnavigation-util/src/main/java/com/mapbox/navigation/utils/internal/LoggerProvider.kt

```diff
@@ -68,3 +105,21 @@ fun logW(msg: String, category: String? = null) {
 fun logE(msg: String, category: String? = null) {
     LoggerProvider.frontend.logE(msg, category)
 }
+
+/**
+ * @param category optional string to identify the source or category of the log message.
+ * @param lazyMsg is a lazy message to log. The lazy message isn't executed if current log level is less verbose than Error.
+ * Noting that the category is appended to the log message to give extra context along with the `[nav-sdk]` parent category.
+ * As an example, this is how the logs would look like `E/Mapbox: [nav-sdk] [ConnectivityHandler] NetworkStatus=ReachableViaWiFi`.
+ */
+inline fun logE(category: String? = null, lazyMsg: () -> String) {
+    if (logLevel().atLeast(LoggingLevel.ERROR)) {
+        logE(lazyMsg(), category)
+    }
+}
+
+/**
+ * Should not be used directly.
+ * Added to support inline calls. Public inline functions can use only public API inside.
+ */
+fun logLevel() = LoggerProvider.frontend.getLogLevel()
```

Comment:
By the way, you can make it `internal` if you add `@PublishedApi` annotation.

Reply:
Amazing! ❤️ 
https://github.com/mapbox/mapbox-navigation-android/pull/6447